### ISSUE TITLE
Corregir modelo de datos

### ISSUE THREAD

**Issue by Wragdan** [Wragdan/chat-api#4]
El modelo de datos deberia ser el siguiente

## Users: 
- Email
- Password
- Role
- Name

_Hasta el momento users esta bien_

## Conversations:
- Participant1 (cambiar nombre para no usar numeros como 1 y 2)
- Participant2 (misma observacion anterior)

## Messages
- Conversation_id
- Sender
- Content (text, body, o lo que queras poner... preferible body)
- Time (hora a la que se envio)

User participa en varias conversations y cada conversation tiene tiene mensajes en ella...

las rutas deben ser de la siguiente manera

| Metodo | Ruta | Descripcion |
|---|---|---
|POST| login |No necesito explicarlo (devuelve los datos del usuario)
|POST|register|Tampoco necesito explicarlo (solo devuelve mensaje de estado)
|POST|conversations| Agrega una nueva conversation
|GET|conversations/:user_id|Devuelve las conversaciones de ese user
|GET|users|Devuele todos los users
|GET|users/:id|Devuelve el user con dicho id
|GET|messages/:conversation_id|Devuelve todos los mensajes de dicha conversation
|POST|messages/:conversation_id|Crea un nuevo mensaje en dicha conversation
|PUT|users/:id|Actualiza los datos de dicho usuario

Por ahora solo eso, los deletes hacelos de ultimo

**Comment by RinerJBonilla** [comment]
![screenshot from 2018-02-27 17-13-50](https://user-images.githubusercontent.com/20601938/36771068-bd1a760c-1c13-11e8-95a4-38ba208e25f4.png)
ahora recipient_id y sender_id son tipo string y se relacionan con el nombre de usuario, y no de su id

**Comment by RinerJBonilla** [comment]
![screenshot from 2018-02-27 17-14-04](https://user-images.githubusercontent.com/20601938/36771088-dbde7e3a-1c13-11e8-94f6-26aadf517ec9.png)
el atributo content de messages ahora se llama body y tiene un nuevo atributo que se llama time que es tipo datetime

**Comment by Wragdan** [comment]
No, me referia a que en lugar de participant1 le pusieras algo como firstParticipant, para nu usar numeros... que no usen el usuario, que usen el id siempre

**Comment by Wragdan** [comment]
y cambia toda esa ruta, escribir **_conversations/:id/messages_** es muy tedioso, hace una ruta **_messages/:conversation_id_**

**Comment by RinerJBonilla** [comment]
![screenshot from 2018-02-28 00-18-05](https://user-images.githubusercontent.com/20601938/36787234-03404270-1c4f-11e8-80d1-45de6910da55.png)
todas las rutas que pediste ya estean como queres, solo el GET **conversations/:user_id es** **conversation/:user_id** ya que conversations/(:numero) va hacia otro metodo

**Comment by RinerJBonilla** [comment]
con lo del model conversations, las columnas nunca han llevado numero, son **sender_id** y **recipient_id**, a menos que no te guste esos nombre y querras **firstparticipant secondparticipant**

**Comment by Wragdan** [comment]
Si no me equivoco, al momento de llamar **conversations/:user_id** o **conversation/:user_id** se escriben de la siguiente manera **conversation/1** , por lo que habra un choque entre las dos rutas...

**Comment by Wragdan** [comment]
> con lo del model conversations, las columnas nunca han llevado numero, son sender_id y recipient_id, a menos que no te guste esos nombre y querras firstparticipant secondparticipant

cuando me refiero a que lleven numero, era a **MI** ejemplo, donde dije 
> participant1

Porque sender no es un nombre que le podria dar a una conversacion.. los llamo participantes porque no me interesa quien es el que la inicio. 

Lo importante es que la ruta **_conversations/:user_id_** me devuelva las conversations donde el user_id sea el sender o el recipient...

**Comment by RinerJBonilla** [comment]
exacto, resources crea **conversations/:numero** que dirige a la conversacion con ese **:id** pero lo que nesecitas era que el numero fuera el **:id** del usuario, y como choca con esa ruta, cree esa nueva con la palbra c**onversation/:numero** en _singular_

**Comment by Wragdan** [comment]
> exacto, resources crea conversations/:numero que dirige a la conversacion con ese :id pero lo que nesecitas era que el numero fuera el :id del usuario, y como choca con esa ruta, cree esa nueva con la palbra conversation/:numero en singular

No te habia entendido, la palabra es **conversation** en singular... ya comprendi...

**Comment by Wragdan** [comment]
Haceme un favor, y desasigna el token required a todas las rutas, por falta de tiempo, no quiero estar validando si envio el token o no...

**Comment by Wragdan** [comment]
> el atributo content de messages ahora se llama body y tiene un nuevo atributo que se llama time que es tipo datetime

No tenias que crear time (se que created_at guarda el valor), pero entiendo tu confusion, al fin y al cabo, yo pedi el cambio.

**Comment by Wragdan** [comment]
Cuando termines estos cambios, subi a tu branch y haces pull request

**Comment by RinerJBonilla** [comment]
listo, ya tiene las nuevas routes y hace skip de authentication

**Comment by Wragdan** [comment]
![image](https://user-images.githubusercontent.com/17775924/36788733-9b5cb9a8-1c54-11e8-9612-ea6da389a435.png)

Sigue pidiendome token

**Comment by Wragdan** [comment]
Also en **messages/:conversation_id** te manda a index. Muestra todos los mensajes o todos los mensajes con esa conversation_id?

**Comment by RinerJBonilla** [comment]
ohhh, hice al reves el pull-request

**Comment by RinerJBonilla** [comment]
**messages/:conversation_id** te lleva al index de todos los mensajes de una conversacion, pero te los tira todos si el user que estas logeado tiene role 'admin'

**Comment by RinerJBonilla** [comment]
ya estan las rutas arregladas en branch:corrections

**Comment by RinerJBonilla** [comment]
solo quiero que comfirmes el modelo conversation si se quede en sender_id:(user.id) y recipient_id:(user.id) o en verdad queres que sea con la palabra participant

**Comment by RinerJBonilla** [comment]
asi ya puedo crear objetos default en db:seeds

**Comment by Wragdan** [comment]
dejalo asi... Pero independientemente de como se llame. la ruta **conversation/:user_id** necesita enviarme los mensajes de la conversacion si el user id es el sender o recipient, ya que son los dos participantes

**Comment by RinerJBonilla** [comment]
ok ya esta, probalo y me decis si funciona la ruta

**Comment by RinerJBonilla** [comment]
solo esta en branch corrections
